src/sentry/api/endpoints/organization_config_integrations.py

```diff
@@ -2,14 +2,26 @@
 
 from rest_framework.response import Response
 
-from sentry import integrations
+from django.conf import settings
+
+from sentry import integrations, features
 from sentry.api.bases.organization import OrganizationEndpoint
 from sentry.api.serializers import serialize, IntegrationProviderSerializer
 
 
 class OrganizationConfigIntegrationsEndpoint(OrganizationEndpoint):
     def get(self, request, organization):
-        providers = list(integrations.all())
+        has_catchall = features.has(
+            "organizations:internal-catchall", organization, actor=request.user
+        )
+
+        providers = []
+        for provider in integrations.all():
+            if not has_catchall and provider.key in settings.SENTRY_INTERNAL_INTEGRATIONS:
```

Comment:
@MeredithAnya Why not just use `.filter`? Always better to be functional when possible :)

Reply:
the IntegrationManager doesn't have a `filter` attribute, and I don't think it's probably super necessary since the feature gating is only temporary till it's released to everyone

Reply:
@MeredithAnya Yea, I guess it wouldn't be an attribute. But you still could do something like:
```
providers = filter(lambda provider: not has_catchall and provider.key in settings.SENTRY_INTERNAL_INTEGRATIONS, integrations.all())
```
I agree in this case it's not super necessary because it is temporary, but in the future, I do think it would be better to be functional even with temporary code if only to get in the habit of it and becoming more comfortable with functional programming in general.